subsys/suit/envelope_info/src/suit_envelope_info.c

```diff
@@ -3,78 +3,108 @@
+#define DFU_PARTITION_EB_SIZE FIXED_PARTITION_ERASE_BLOCK_SIZE(DFU_PARTITION_LABEL)
+#define DFU_PARTITION_WB_SIZE FIXED_PARTITION_WRITE_BLOCK_SIZE(DFU_PARTITION_LABEL)
+#define DFU_PARTITION_DEVICE  FIXED_PARTITION_DEVICE(DFU_PARTITION_LABEL)
+#define EMPTY_STORAGE_VALUE   0xFFFFFFFF
 
-static suit_plat_err_t update_candidate_address_and_size_validate(const uint8_t *addr, size_t size)
+suit_plat_err_t suit_dfu_partition_envelope_info_get(const uint8_t **address, size_t *size)
 {
-	if (addr == NULL || *((uint32_t *)addr) == EMPTY_STORAGE_VALUE) {
-		LOG_DBG("Invalid update candidate address: %p", (void *)addr);
-		return SUIT_PLAT_ERR_INVAL;
-	}
 
-	if (size == 0) {
-		LOG_DBG("Invalid update candidate size: %d", size);
+	zcbor_state_t states[3];
+
+	struct nvm_address device_offset = {
+		.fdev = DFU_PARTITION_DEVICE,
+		.offset = DFU_PARTITION_OFFSET,
+	};
+
+	if (address == NULL || size == NULL) {
+		LOG_ERR("Wrong parameters");
 		return SUIT_PLAT_ERR_INVAL;
 	}
 
-	return SUIT_PLAT_SUCCESS;
-}
+	uintptr_t mapped_address = 0;
 
-suit_plat_err_t suit_envelope_info_candidate_stored(const uint8_t *address, size_t max_size)
-{
-	zcbor_state_t states[3];
+	if (!suit_memory_nvm_address_to_global_address(&device_offset, &mapped_address)) {
+		LOG_ERR("Cannot obtain memory-mapped DFU Partition address");
+		return SUIT_PLAT_ERR_IO;
+	}
 
-	if (address == NULL) {
-		return SUIT_PLAT_ERR_INVAL;
+	const uint8_t *envelope_address = (uint8_t *)mapped_address;
+	size_t envelope_size = DFU_PARTITION_SIZE;
+
+	if (*((uint32_t *)envelope_address) == EMPTY_STORAGE_VALUE) {
+		LOG_DBG("DFU Partition empty");
+		return SUIT_PLAT_ERR_NOT_FOUND;
 	}
 
-	zcbor_new_state(states, sizeof(states) / sizeof(zcbor_state_t), address, max_size, 1, NULL,
-			0);
+	zcbor_new_state(states, sizeof(states) / sizeof(zcbor_state_t), envelope_address,
+			envelope_size, 1, NULL, 0);
 
 	/* Expect SUIT envelope tag (107) and skip until the end of the envelope */
 	if (!zcbor_tag_expect(states, 107) || !zcbor_any_skip(states, NULL)) {
 		LOG_DBG("Malformed envelope");
 		return SUIT_PLAT_ERR_CBOR_DECODING;
 	}
 
-	envelope_size = MIN(max_size, (size_t)states[0].payload - (size_t)address);
-	envelope_address = address;
+	envelope_size = (size_t)states[0].payload - (size_t)envelope_address;
+
+	if (envelope_size > DFU_PARTITION_SIZE) {
+		LOG_DBG("Malformed envelope");
+		return SUIT_PLAT_ERR_CBOR_DECODING;
+	}
+
+	*address = envelope_address;
+	*size = envelope_size;
 
 	return SUIT_PLAT_SUCCESS;
 }
 
-void suit_envelope_info_reset(void)
+suit_plat_err_t suit_dfu_partition_device_info_get(struct suit_nvm_device_info *device_info)
 {
-	envelope_size = 0;
-	envelope_address = NULL;
-}
+	if (device_info == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
 
-suit_plat_err_t suit_envelope_info_get(const uint8_t **address, size_t *size)
-{
-	suit_plat_err_t ret = SUIT_PLAT_ERR_INVAL;
+	device_info->fdev = DFU_PARTITION_DEVICE;
```

Comment:
It might be worth checking at this stage if `device_is_ready(device_info->fdev)`.